File: NotesApp/NotesApp/Models/Note.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace NotesApp.Models;

public class Note
{
    [Key]
    public int Id { get; set; }

    [Required]
    [MaxLength(200)]
    public string Title { get; set; } = string.Empty;

    public string Content { get; set; } = string.Empty;

    [MaxLength(260)]
    public string? MarkdownFileName { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.Now;

    public DateTime UpdatedAt { get; set; } = DateTime.Now;

    public bool IsPinned { get; set; }

    public bool IsDeleted { get; set; }

    public ICollection<NoteTag> NoteTags { get; set; } = new List<NoteTag>();
}

File: Sources/NotesCore/Models/Note.swift
import Foundation

public struct Note: Identifiable, Hashable {
    public var id: Int64
    public var title: String
    public var content: String
    public var markdownFileName: String?
    public var createdAt: Date
    public var updatedAt: Date
    public var isPinned: Bool
    public var isDeleted: Bool
    public var tags: [Tag]

    public init(
        id: Int64,
        title: String,
        content: String,
        markdownFileName: String?,
        createdAt: Date,
        updatedAt: Date,
        isPinned: Bool,
        isDeleted: Bool,
        tags: [Tag] = []
    ) {
        self.id = id
        self.title = title
        self.content = content
        self.markdownFileName = markdownFileName
        self.createdAt = createdAt
        self.updatedAt = updatedAt
        self.isPinned = isPinned
        self.isDeleted = isDeleted
        self.tags = tags
    }
}
